src/plugins/discover/public/application/main/components/sidebar/discover_field.tsx

```diff
@@ -29,7 +29,7 @@ import type { DataViewField, DataView } from '@kbn/data-views-plugin/public';
 import { getTypeForFieldIcon } from '../../../../utils/get_type_for_field_icon';
 import { DiscoverFieldDetails } from './discover_field_details';
 import { FieldDetails } from './types';
-import { getFieldTypeName } from './lib/get_field_type_name';
```

Comment:
Thx a lot for cleaning this up and removing duplicated code 👍 . You can flag the PR ready for review. One thing you could add on top of it, could be a Jest test for e.g. `discover_field.tsx `that would catch the error. Or a basic one for `get_field_type_name.ts`